<!-- Launch extraction of planes and table detection together with visualization -->
<launch>

    <!-- The topic of the input cloud to segment into planes and detect tables -->
    <arg name="cloud" default="/head_xtion/depth/points"/>
    <!-- The topic to publish the detected tables on -->
    <arg name="detection_topic" value="/table_detection/table_primitives"/>
    <!-- The topic to publish the resulting found planes on -->
    <arg name="primitive_topic" value="/primitive_extraction/primitives"/>
    <!-- The topic to publish the tracked tables on -->
    <arg name="table_topic" value="/table_detection/tables"/>

    <!-- Launch the primitive extraction, only extract planes -->
    <include file="$(find primitive_extraction)/launch/extraction.launch">
        <arg name="cloud" value="$(arg cloud)"/>
        <arg name="primitive_topic" value="$(arg primitive_topic)"/>
        <arg name="extract_cylinders" value="false"/>
        <arg name="extract_spheres" value="false"/>
        <arg name="min_height" value="0.5"/>
        <arg name="max_height" value="2.0"/>
    </include>
    
    <!-- Visualize the detected tables with a marker array -->
    <node pkg="primitive_extraction" type="primitives_to_markers" name="tables_to_markers" output="screen">
        <param name="input" type="string" value="$(arg detection_topic)"/>
        <param name="output" type="string" value="/table_detection/primitive_marker_array"/>
        <param name="display_plane_box" type="bool" value="false"/>
        <param name="plane_r" type="double" value="0.0"/>
        <param name="plane_g" type="double" value="0.0"/>
        <param name="plane_b" type="double" value="1.0"/>
    </node>
    
</launch>
